docs/rfcs/0011-decouple-home-activity-and-external-activity.md

```diff
@@ -0,0 +1,165 @@
+
+4. The `ExternalAppBrowserActivity` currently uses the `HomeActivity` `onCreate()` implementation. This forces the Custom Tabs features to to perform unnecessary checks such as:
+* `maybeShowSplashScreen()`
+* `shouldShowOnboarding()`
+* perform checks for Homepage Contile feature activity
+* perform checks for Pocket feature activity
+
+This list is not exhaustive but gives an insight into the redundant work being forced upon the `ExternalAppBrowserActivity`. This applies to many of the other `HomeActivity` lifecycle related functions.
+
+## Proposal
+Reduce the coupling of the `Activity`s. Establish `HomeActivity` as the *main* `Activity` and define clear responsibilities. Below is an list of suggested refactorings to initiate the decoupling.
+
+1. Remove the no-ops from `ExternalAppBrowserActivity`. [Example PR.](https://github.com/mozilla-mobile/firefox-android/pull/4851)
+
+2. Create an `Activity` 'navigation' file that takes an `Activity` to delegate the `Activity`s `getNavDirections()` functionality. This removes the definition from `HomeActivity` and the requirement to override it in `ExternalAppBrowserActivity`. [Example PR.](https://github.com/mozilla-mobile/firefox-android/pull/4854)
+
+3. Create a `Fragment` 'navigation' file to delegate the `HomeActivity` 'open to browser' functionality to. This removes another responsibility from `HomeActivity`. [Example PR.](https://github.com/mozilla-mobile/firefox-android/pull/4857)
+
+4. Extract the `getBreadcrumbMessage`, `getIntentSource` and `getIntentSessionId` functions from `HomeActivity` using an `Activity` delegate. This removes the definition from `HomeActivity` and the requirement to override it in `ExternalAppBrowserActivity`. [Example PR.](https://github.com/mozilla-mobile/firefox-android/pull/4872)
+
+5. Update `HomeActivity` & `ExternalAppBrowserActivity` with the necessary visibility modifiers to enforce constraints.
+
+6. Move the `handleRequestDesktopMode` function to another file. This removes another responsibility from `HomeActivity`.
+
+7. Rename `HomeActivity` to be explicit that this is the *main* `Activity`. For info, the Focus application uses a `MainActivity`, the Reference Browser application uses a `BrowserActivity`.
+
+The next steps are less cleary defined and require more investigation on the completion of the above tasks.
+
+* Explore alternatives to the current Inheritance structure. E.g. Kotlin Delegation methods, or an alternative 'base' `Activity` model.
+* Prevent unnecessary checks being carried out by `ExternalAppBrowserActivity` in `HomeActivity` lifecycle related functions.
+
+## Drawbacks
+* Current reported Custom Tabs Bugzilla bugs may be affected by the changes mentioned here.
+* No immediate tangible user facing improvements.
+
+## Rationale and alternatives
+### Rationale 
+* It is well established best practice to ['Prefer composition over inheritence'](https://en.wikipedia.org/wiki/Composition_over_inheritance).
```

Comment:
This is a building design block of AC as well and why we write all our feature classes that way. Here is [an example](https://github.com/mozilla-mobile/reference-browser/pull/2338) where we are moving some reusable code into that pattern as well.

We can then upstream the change with easily if it's useful.